src/pykeen/evaluation/evaluator.py

```diff
@@ -381,16 +377,13 @@ def _param_size_search(
         return cast(Tuple[int, bool], (values_dict[key], evaluated_once))
 
     @staticmethod
-    def _check_slicing_availability(model: Model, batch_size: int) -> None:
+    def _check_slicing_availability(model: Model, batch_size: int, entities: bool, relations: bool) -> None:
         # Test if slicing is implemented for the required functions of this model
-        if model.use_inverse_triples:
-            if not model.can_slice_t:
-                raise MemoryError(
-                    f"The current model can't be evaluated on this hardware with these parameters, as "
-                    f"evaluation batch_size={batch_size} is too big and slicing is not implemented for "
-                    f"this model yet."
-                )
-        elif not model.can_slice_t or not model.can_slice_h:
+        if any(
+            (entities and model.use_inverse_triples and not model.can_slice_t)
```

Comment:
might be helpful to save each of these 3 conditions into variables first, such that the variable names provide some implicit documentation

Reply:
d72895e